core/internal/features_test.go

```diff
@@ -1216,3 +1230,223 @@ func TestIntegration_MultiwordV1_Sim(t *testing.T) {
+	return owner, b, ocrContractAddress, ocrContract
+}
+
+func setupNode(t *testing.T, owner *bind.TransactOpts, port int, dbName string, b *backends.SimulatedBackend) (*cltest.TestApplication, string, common.Address, ocrkey.EncryptedKeyBundle, func()) {
+	config, _, ormCleanup := cltest.BootstrapThrowawayORM(t, fmt.Sprintf("%s%s", dbName, strings.Replace(uuid.New(), "-", "", -1)), true)
+	config.Dialect = orm.DialectPostgresWithoutLock
+	app, appCleanup := cltest.NewApplicationWithConfigAndKeyOnSimulatedBlockchain(t, config, b)
+	_, _, err := app.Store.OCRKeyStore.GenerateEncryptedP2PKey()
+	require.NoError(t, err)
+	p2pIDs := app.Store.OCRKeyStore.DecryptedP2PKeys()
+	require.NoError(t, err)
+	require.Len(t, p2pIDs, 1)
+	peerID := p2pIDs[0].MustGetPeerID().String()
+
+	app.Config.Set("P2P_PEER_ID", peerID)
+	app.Config.Set("P2P_LISTEN_PORT", port)
+	app.Config.Set("ETH_HEAD_TRACKER_MAX_BUFFER_SIZE", 100)
+	app.Config.Set("MIN_OUTGOING_CONFIRMATIONS", 1)
+	app.Config.Set("CHAINLINK_DEV", true) // Disables ocr spec validation so we can have fast polling for the test.
+
+	transmitter := app.Store.KeyStore.Accounts()[0].Address
+
+	// Fund the transmitter address with some ETH
+	n, err := b.NonceAt(context.Background(), owner.From, nil)
+	require.NoError(t, err)
+
+	tx := types.NewTransaction(n, transmitter, big.NewInt(1000000000000000000), 21000, big.NewInt(1), nil)
+	signedTx, err := owner.Signer(owner.From, tx)
+	require.NoError(t, err)
+	err = b.SendTransaction(context.Background(), signedTx)
+	require.NoError(t, err)
+	b.Commit()
+
+	_, kb, err := app.Store.OCRKeyStore.GenerateEncryptedOCRKeyBundle()
+	require.NoError(t, err)
+	return app, peerID, transmitter, kb, func() {
+		ormCleanup()
+		appCleanup()
+	}
+}
+
+func TestIntegration_OCR(t *testing.T) {
+	owner, b, ocrContractAddress, ocrContract := setupOCRContracts(t)
+
+	// Note it's plausible these ports could be occupied on a CI machine.
+	// May need a port randomize + retry approach if we observe collisions.
+	appBootstrap, bootstrapPeerID, _, _, cleanup := setupNode(t, owner, 19999, "bootstrap", b)
+	defer cleanup()
+
+	var (
+		oracles      []confighelper.OracleIdentity
+		transmitters []common.Address
+		kbs          []ocrkey.EncryptedKeyBundle
+		apps         []*cltest.TestApplication
+	)
+	for i := 0; i < 4; i++ {
+		app, peerID, transmitter, kb, cleanup := setupNode(t, owner, 20000+i, fmt.Sprintf("oracle%d", i), b)
+		defer cleanup()
+
+		kbs = append(kbs, kb)
+		apps = append(apps, app)
+		transmitters = append(transmitters, transmitter)
+
+		oracles = append(oracles, confighelper.OracleIdentity{
+			OnChainSigningAddress:           ocrtypes.OnChainSigningAddress(kb.OnChainSigningAddress),
+			TransmitAddress:                 transmitter,
+			OffchainPublicKey:               ocrtypes.OffchainPublicKey(kb.OffChainPublicKey),
+			PeerID:                          peerID,
+			SharedSecretEncryptionPublicKey: ocrtypes.SharedSecretEncryptionPublicKey(kb.ConfigPublicKey),
+		})
```

Comment:
You may want to call `peer.Close()` here

Reply:
app.Stop() might also do it

Reply:
spoke offline fixed in https://github.com/smartcontractkit/chainlink/pull/3856/files